apple/testing/ios_rules.bzl

```diff
@@ -175,6 +175,9 @@ def _ios_test(
         target_name_template = "%s_test_binary",
     )
 
+    if "provisioning_profile" in kwargs:
```

Comment:
if you use pop instead of get, you don't need to add the attribute to the test rules above, which in any case do not get processed there.

Reply:
also, can you limit this to `ios_ui_test` only? `ios_unit_test` does not require a provisioning profile

Reply:
> also, can you limit this to `ios_ui_test` only? `ios_unit_test` does not require a provisioning profile

Have you any suggestion for separating them? I have no idea where should I write it in `_ios_test` or `binary_support.create_linked_binary_target`
It seems that rules depend `entitlements` for sign and in my case we don't need to add entitlements.

Reply:
you can add an argument to `_ios_test` with the name `uses_provisioning_profile` and then use 

```
if uses_provisioning_profile:
    bundling_args["provisioning_profile"] = kwargs.pop("provisioning_profile", None)
```

and then add `uses_provisioning_profile = True` on `ios_ui_test` below

This way, the `ios_unit_test` will fail if the `provisioning_profile` attribute is passed, and on `ios_ui_test` tests the attribute will be filtered and only passed to the intermediate `_ios_test_bundle` target

Reply:
> you can add an argument to `_ios_test` with the name `uses_provisioning_profile` and then use
> 
> ```
> if uses_provisioning_profile:
>     bundling_args["provisioning_profile"] = kwargs.pop("provisioning_profile", None)
> ```
> and then add `uses_provisioning_profile = True` on `ios_ui_test` below
> 
> This way, the `ios_unit_test` will fail if the `provisioning_profile` attribute is passed, and on `ios_ui_test` tests the attribute will be filtered and only passed to the intermediate `_ios_test_bundle` target

Modified and rebased